### PR TITLE
Add regression tests for scatter 1D tensor negative dim support

### PR DESCRIPTION
**Pull request by cmaryanTT**
## Summary
- Add test coverage for 1D tensors with negative dimensions in ttnn.scatter to prevent regression of the fix in PR #41762
- Add `(100,)` 1D shape to `test_scatter_negative_dim` parametrization with skip logic for invalid dim combinations
- Add `((100,), -1)` case to `test_scatter_negative_dim_equals_positive`
- Add dedicated `test_scatter_1d_tile_layout_negative_dim` test that specifically covers 1D tensors in TILE_LAYOUT with `dim=-1`

## Background
PR #41762 fixed a bug where dim normalization used `padded_shape().rank()` instead of `logical_shape().rank()`. For 1D tensors in TILE_LAYOUT, the padded shape has rank 2 (e.g., `[100]` → `[1, 128]`), causing `dim=-1` to normalize incorrectly to `1` instead of `0`.

## Test plan
- [x] `test_scatter_1d_tile_layout_negative_dim` passes (2 tests)
- [x] `test_scatter_negative_dim` passes with 1D shapes (14 tests pass, 2 skip for invalid dim=-2)
- [x] `test_scatter_negative_dim_equals_positive` passes with 1D shape (1 new test)

### CI Status
_Auto-generated on every push. Badges update live. Click a pipeline name or badge to filter by this branch._

| Pipeline | Status | Latest Run |
|---|:---:|---|
| [Sanity tests](https://github.com/tenstorrent/tt-metal/actions/workflows/sanity-tests.yaml?query=branch:test/scatter-gather-1d-negative-dim-regression) | [](https://github.com/tenstorrent/tt-metal/actions/workflows/sanity-tests.yaml?query=branch:test/scatter-gather-1d-negative-dim-regression) | [Launch ↗](https://github.com/tenstorrent/tt-metal/actions/workflows/sanity-tests.yaml?query=branch:test/scatter-gather-1d-negative-dim-regression) |
| [Blackhole post-commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml?query=branch:test/scatter-gather-1d-negative-dim-regression) | [](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml?query=branch:test/scatter-gather-1d-negative-dim-regression) | [Launch ↗](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml?query=branch:test/scatter-gather-1d-negative-dim-regression) |
| [L2 Nightly](https://github.com/tenstorrent/tt-metal/actions/workflows/tt-metal-l2-nightly.yaml?query=branch:test/scatter-gather-1d-negative-dim-regression) | [](https://github.com/tenstorrent/tt-metal/actions/workflows/tt-metal-l2-nightly.yaml?query=branch:test/scatter-gather-1d-negative-dim-regression) | [Launch ↗](https://github.com/tenstorrent/tt-metal/actions/workflows/tt-metal-l2-nightly.yaml?query=branch:test/scatter-gather-1d-negative-dim-regression) |
| [(Single) Choose your pipeline](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select.yaml?query=branch:test/scatter-gather-1d-negative-dim-regression) | [](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select.yaml?query=branch:test/scatter-gather-1d-negative-dim-regression) | [Launch ↗](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select.yaml?query=branch:test/scatter-gather-1d-negative-dim-regression) |
| [(T3K) Choose your pipeline](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select-t3k.yaml?query=branch:test/scatter-gather-1d-negative-dim-regression) | [](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select-t3k.yaml?query=branch:test/scatter-gather-1d-negative-dim-regression) | [Launch ↗](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select-t3k.yaml?query=branch:test/scatter-gather-1d-negative-dim-regression) |
| [(Galaxy) Choose your pipeline](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select-galaxy.yaml?query=branch:test/scatter-gather-1d-negative-dim-regression) | [](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select-galaxy.yaml?query=branch:test/scatter-gather-1d-negative-dim-regression) | [Launch ↗](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select-galaxy.yaml?query=branch:test/scatter-gather-1d-negative-dim-regression) |

### CODE REVIEW

File: tests/ttnn/nightly/unit_tests/operations/data_movement/test_scatter.py [hunk 1]
```diff
@@ -503,14 +503,19 @@ def test_scatter_forge(input_shape, index_and_source_shape, input_dtype, device)
...
 def test_scatter_negative_dim(device, shape, dim, dtype):
     """
     Regression test for negative dimension support in ttnn.scatter.
     Verifies that negative dimension values work correctly and match PyTorch behavior.
+    Includes 1D tensors to cover the logical_shape vs padded_shape bug (PR #41762).
     """
+    # Skip invalid dim combinations (e.g., dim=-2 for 1D tensor)
+    if abs(dim) > len(shape):
+        pytest.skip(f"dim={dim} invalid for rank={len(shape)} tensor")
+
```
Comment (jvegaTT) [inline tests/ttnn/nightly/unit_tests/operations/data_movement/test_scatter.py:518]: Nit. This logic wouldn't work for +dim tensors (dim 2 on a 2d tensor is invalid but this would not skip). However since the test is only for negative dims this is fine.

File: tests/ttnn/nightly/unit_tests/operations/data_movement/test_scatter.py [hunk 2]
```diff
@@ -590,3 +597,43 @@ def test_scatter_negative_dim_equals_positive(device, shape, dim):
...
+    The bug was that dim normalization used padded_shape().rank() instead of
+    logical_shape().rank(). For a 1D tensor [100] in TILE_LAYOUT, the padded
+    shape becomes [1, 128] (rank 2), causing dim=-1 to normalize to 1 instead
+    of 0, which then fails validation.
+    """
+    torch.manual_seed(0)
+    torch_dtype = torch.bfloat16
+
+    torch_input = torch.randn(shape, dtype=torch_dtype)
+    torch_index = torch.randint(0, shape[0], index_shape, dtype=torch.int64)
```
Comment (jvegaTT) [inline tests/ttnn/nightly/unit_tests/operations/data_movement/test_scatter.py:623]: Nit. should we use torch.int32?

**Review by jvegaTT — APPROVED** [review]
Just a couple non blocking minor nits
